modules/detection/NoAudioSignalDetection.js

```diff
@@ -0,0 +1,155 @@
+import EventEmitter from 'events';
+import * as JitsiConferenceEvents from '../../JitsiConferenceEvents';
+import * as DetectionEvents from './DetectionEvents';
+
+// We wait a certain time interval for constant silence input from the current device to account for
+// potential abnormalities and for a better use experience i.e. don't generate event the instant
+// an audio track is added to the tcr.
+// Potential improvement - add this as a configurable parameter.
+const SILENCE_PERIOD_MS = 4000;
+
+/**
+ * Detect if there is no audio input on the current TraceAblePeerConnection selected track. The no audio
+ * state must be constant for a configured amount of time in order for the event to be triggered.
+ * @fires DetectionEvents.AUDIO_INPUT_STATE_CHANGE
+ * @fires DetectionEvents.NO_AUDIO_INPUT
+ */
+export default class NoAudioSignalDetection extends EventEmitter {
+    /**
+     * @param conference the JitsiConference instance that created us.
+     * @constructor
+     */
+    constructor(conference) {
+        super();
+
+        this._conference = conference;
+        this._timeoutTrigger = null;
+        this._hasAudioInput = null;
```

Comment:
Not really, I just enjoy consistency and because I initialise them with null on cleanup I though I'd do the same thing here.
And from what I saw throughout the code you guys define the class members in the c'tor for consistency and readability I guess.. and initialise them with null.

Reply:
i.e. RTC.js and others.

Reply:
I personally always prefer to use `undefined`. We don't have a rule for this tho.

I also haven't noticed that we try to define all the members in the constructor. Actually if possible to define something I will always omit it :)